packages/worker/src/utilities/email.ts

```diff
@@ -165,39 +163,16 @@ export async function sendEmail(
     }),
   }
   if (opts?.attachments) {
-    const attachments = await Promise.all(
-      opts.attachments?.map(async attachment => {
-        const isFullyFormedUrl =
-          attachment.url.startsWith("http://") ||
-          attachment.url.startsWith("https://")
-        if (isFullyFormedUrl) {
-          const response = await fetch(attachment.url)
-          if (!response.ok) {
-            throw new Error(`unexpected response ${response.statusText}`)
-          }
-          const fallbackFilename = path.basename(
-            new URL(attachment.url).pathname
-          )
-          return {
-            filename: attachment.filename || fallbackFilename,
-            content: response?.body,
-          }
-        } else {
-          const url = attachment.url
-          const result = objectStore.extractBucketAndPath(url)
-          if (result === null) {
-            throw new Error("Invalid signed URL")
-          }
-          const { bucket, path } = result
-          const readStream = await objectStore.getReadStream(bucket, path)
-          const fallbackFilename = path.split("/").pop() || ""
-          return {
-            filename: attachment.filename || fallbackFilename,
-            content: readStream,
-          }
-        }
-      })
+    let attachments = await Promise.all(
+      opts.attachments?.map(objectStore.processAutomationAttachment)
     )
+    attachments = attachments.map(attachment => {
+      if ("path" in attachment) {
+        const { path, ...rest } = attachment
+        return rest
+      }
+      return attachment
+    })
```

Comment:
```suggestion
    attachments = attachments.map(attachment => {
      return _.omit(attachment, ["path"])
    })
```

https://lodash.com/docs/4.17.15#omit